Tests/ResponseSerializationTests.swift

```diff
@@ -567,6 +592,67 @@ final class DecodableResponseSerializerTests: BaseTestCase {
         XCTAssertNotNil(result.value)
         XCTAssertNil(result.error)
     }
+
+    func testThatDecodableResponseSerializerSucceedsWhenDataIsNilWithEmptyResponseConformingTypeAndEmptyResponseStatusCode() {
+        // Given
+        let serializer = DecodableResponseSerializer<Bool>()
+        let response = HTTPURLResponse(statusCode: 204)
+
+        // When
+        let result = Result { try serializer.serialize(request: nil, response: response, data: nil, error: nil) }
+
+        // Then
+        XCTAssertTrue(result.isSuccess)
+        XCTAssertNotNil(result.value)
+        XCTAssertNil(result.error)
+    }
+
+    func testThatDecodableResponseSerializerFailsWhenDataIsNilWithEmptyResponseNonconformingTypeAndEmptyResponseStatusCode() {
+        // Given
+        let serializer = DecodableResponseSerializer<Int>()
+        let response = HTTPURLResponse(statusCode: 204)
+
+        // When
+        let result = Result { try serializer.serialize(request: nil, response: response, data: nil, error: nil) }
+
+        // Then
+        XCTAssertTrue(result.isFailure)
+        XCTAssertNil(result.value)
+        XCTAssertNotNil(result.error)
+
+        if let error = result.error?.asAFError {
+            XCTAssertTrue(error.isInvalidEmptyResponse)
+        } else {
+            XCTFail("error should not be nil")
+        }
+    }
+
+    final class DecodableClassWithEmptyResponse: Decodable, EmptyResponse {
+        static func emptyResponseValue() -> DecodableClassWithEmptyResponse {
+            return emptyValue
+        }
+
+        static let emptyValue = DecodableClassWithEmptyResponse(string: "");
+
+        init(string: String) { self.string = string }
+
+        let string: String
+    }
+
+    func testThatJSONDecodableResponseSerializerSucceedsWhenDataIsNilWithEmptyResponseStatusCodeAndDecodableClassHasEmptyResponse() {
```

Comment:
What is this testing? It seems unnecessary with all of the other tests around the feature.

Reply:
I was intending to specifically test for identity in case the case of a class being used. Removed in 53775470973f7dd57c2ea4a04e7cd2b8595fcfc8.